packages/ethereum-contracts/contracts/agreements/CFAv1Forwarder.sol

```diff
@@ -0,0 +1,493 @@
+    IConstantFlowAgreementV1 internal _cfa;
+
+    // is tied to a specific instance of host and agreement contracts at deploy time
+    constructor(ISuperfluid host) {
+        _host = host;
+        _cfa = IConstantFlowAgreementV1(address(_host.getAgreementClass(
+            keccak256("org.superfluid-finance.agreements.ConstantFlowAgreement.v1")
+        )));
+    }
+
+    /**
+     * @notice Sets the given flowrate between msg.sender and a given receiver.
+     * If there's no pre-existing flow and `flowrate` non-zero, a new flow is created.
+     * If there's an existing flow and `flowrate` non-zero, the flowrate of that flow is updated.
+     * If there's an existing flow and `flowrate` zero, the flow is deleted.
+     * If the existing and given flowrate are equal, no action is taken.
+     * On creation of a flow, a "buffer" amount is automatically detracted from the sender account's available balance.
+     * This buffer is redeemed on deletion of that flow, if the sender account is then solvent (non-zero balance).
+     * @param token Super token address
+     * @param receiver The receiver of the flow
+     * @param flowrate The wanted flowrate in wad/second. Only positive values are valid here.
+     */
+    function setFlowrate(ISuperToken token, address receiver, int96 flowrate) external {
+       _setFlowrateFrom(token, msg.sender, receiver, flowrate);
+    }
+
+    /**
+     * @notice Like `setFlowrate`, but can be invoked by an account with flowOperator permissions
+     * on behalf of the sender account.
+     */
+    function setFlowrateFrom(ISuperToken token, address sender, address receiver, int96 flowrate) external {
+        _setFlowrateFrom(token, sender, receiver, flowrate);
+    }
+
+    /**
+     * @notice Get the flowrate of the flow between 2 accounts if exists.
+     * @dev Currently, only 0 or 1 flows can exist between 2 accounts. This may change in the future.
+     * @param token Super token address
+     * @param sender The sender of the flow
+     * @param receiver The receiver of the flow
+     * @return flowrate The flowrate from the sender to the receiver account. Returns 0 if no flow exists.
+     */
+    function getFlowrate(ISuperToken token, address sender, address receiver) external view
+        returns(int96 flowrate)
+    {
+        (, flowrate, , ) = _cfa.getFlow(token, sender, receiver);
+    }
+
+    /**
+     * @notice Get all available information about a flow (if exists).
+     * If only the flowrate is needed, consider using `getFlowrate` instead.
+     * @param token Super token address
+     * @param sender The sender of the flow
+     * @param receiver The receiver of the flow
+     * @return lastUpdated Timestamp of last update (flowrate change) or zero if no flow exists
+     * @return flowrate Current flowrate of the flow or zero if no flow exists
+     * @return deposit Deposit amount locked. Returned to the the flow is deleted while solvent. TODO: rename to buffer?
```

Comment:
just a reminder there is a TODO here
maybe this is a good place to start changing the terminology to buffer in our contracts

Reply:
right, I decided to stick to the current names and introduce "buffer" only in the param description